src/plugins/management/public/mocks/index.ts

```diff
@@ -17,10 +17,20 @@
  * under the License.
  */
 
+const createSetupContract = () => ({
```

Comment:
This causes the following to fail:

```
const managementSetup = managementSetupMock.createSetupContract();
managementSetup.sections.getSection.mockReturnValue(foo);
```

Because `managementSetup.sections` is of type [`SectionsServiceSetup`](https://github.com/elastic/kibana/blob/9282f19bf54df2d2ccdfaaf526626e94f29cc45f/src/plugins/management/public/types.ts#L34), rather than `jest.Mocked<SectionsServiceSetup>`.

Do you know of a way to tell TS what we really want to do here? I couldn't get it to work, but didn't spend more than 15 minutes on it.

Reply:
Hmm, I see. IIRC Platform team used something like `DeeplyMockedKeys` instead of `jest.Mocked` for cases like this, maybe that would work for us too... But feel free to ignore if it takes too much time to fix Management's mocks and types.

Reply:
That worked perfectly, thanks for the tip!